File: backend/components/metiers/administration_publique/tests/utils/i18n/test_i18n.py
"""
Tests avancés d'internationalisation pour Threed (multi-langue, fallback, conformité)
"""
import pytest
from ..i18n import i18n
from .i18n_utils import translate

def i18n(msg, lang):
    if lang in ['fr', 'en']:
        return translate(msg, lang)
    return msg

def test_i18n_fr():
    assert '[FR]' in i18n('Test', 'fr')

def test_i18n_en():
    assert '[EN]' in i18n('Test', 'en')

def test_i18n_fallback():
    assert i18n('Test', 'es') == 'Test'

File: backend/components/metiers/administration_publique/tests/utils/i18n/test_i18n.js
// test_i18n.js – Tests JS avancés d'internationalisation pour Threed
const i18n = require('../utils/i18n');

describe('i18n Threed', () => {
  it('doit traduire en français', () => {
    expect(i18n.i18n('Test', 'fr')).toMatch('[FR]');
  });
  it('doit traduire en anglais', () => {
    expect(i18n.i18n('Test', 'en')).toMatch('[EN]');
  });
  it('doit fallback sur la valeur brute', () => {
    expect(i18n.i18n('Test', 'es')).toBe('Test');
  });
});
